com.dynamo.cr/com.dynamo.cr.bob/src/com/defold/extension/pipeline/texture/ITextureCompressor.java

```diff
@@ -28,4 +28,8 @@ public interface ITextureCompressor {
     public default int getAlignedWidth(TextureImage.TextureFormat format, int width) { return width; }
 
     public default int getAlignedHeight(TextureImage.TextureFormat format, int height) { return height; }
+
+    public boolean supportsTextureFormat(TextureImage.TextureFormat format);
+
+    public boolean supportsTextureCompressorPreset(TextureCompressorPreset preset);
```

Comment:
The new functions are here so we can validate presets and formats outside of bob. It's up to the compressor itself to tell the editor (and bob) that it can produce a valid result based on the input.